Panic occurs when running pulumi preview or pulumi up after upgrading to v3.90.0

### What happened?

After upgrading to Pulumi v3.90.0, running pulumi preview or pulumi up results in a panic error. Rolling back to v3.89.0 resolves the issue, suggesting this is an update-related bug.

### Example

```
pulumi preview > out.x
panic: fatal: A failure has occurred: unexpected value type <nil>

goroutine 24 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Failf({0x2adf5d7?, 0xc003225ce0?}, {0xc00253d710?, 0x6?, 0x8?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/fail.go:32 +0xcb
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.object.toDecryptedPropertyValue({{0x0?, 0x0?}, 0xf0?}, {0x32739a0?, 0xc002284b40?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:551 +0x285
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.object.toDecryptedPropertyValue({{0x26304c0?, 0xc003225ce0?}, 0xc8?}, {0x32739a0, 0xc002284b40})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:547 +0x486
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.Map.AsDecryptedPropertyMap(0xc00253db60?, {0x32739a0, 0xc002284b40})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/map.go:87 +0x173
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*evalSource).Iterate(0xc002325020, {0x32884e0?, 0xc0031e5bc0?}, {{0x7fef9ad46978, 0xc00231f840}, 0x7fffffff, 0x0, 0x0, {{0x487e800, 0x0, ...}, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/resource/deploy/source_eval.go:123 +0x2af
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*deploymentExecutor).Execute(0xc008c357b8, {0x32884e0?, 0xc0031e5bc0}, {{0x7fef9ad46978, 0xc00231f840}, 0x7fffffff, 0x0, 0x0, {{0x487e800, 0x0, ...}, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/resource/deploy/deployment_executor.go:167 +0x2b0
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*Deployment).Execute(...)
        /home/runner/work/pulumi/pulumi/pkg/resource/deploy/deployment.go:578
github.com/pulumi/pulumi/pkg/v3/engine.(*deployment).run.func1()
        /home/runner/work/pulumi/pulumi/pkg/engine/deployment.go:314 +0x2b0
created by github.com/pulumi/pulumi/pkg/v3/engine.(*deployment).run in goroutine 1
        /home/runner/work/pulumi/pulumi/pkg/engine/deployment.go:299 +0x29b
```

### Output of `pulumi about`

CLI          
Version      3.90.0
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in nodejs: executable='/home/username/.nvm/versions/node/v18.12.0/bin/node' version='v18.12.0'

Current Stack: stack-name

TYPE
pulumi:pulumi:Stack
pulumi:providers:kubernetes
kubernetes:core/v1:Namespace
kubernetes:helm.sh/v3:Chart
kubernetes:storage.k8s.io/v1:StorageClass
kubernetes:cert-manager.io/v1:ClusterIssuer
kubernetes:core/v1:Secret
kubernetes:networking.k8s.io/v1:Ingress

Found no pending operations associated with stack-name

Backend        
Name           ubuntu22
URL            s3://bucket?awssdk=v2
User           username
Organizations  
Token type     personal

Pulumi locates its logs in /tmp by default

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20231024--cli-config--dont-crash-on-empty-config-values.yaml
+++ b/changelog/pending/20231024--cli-config--dont-crash-on-empty-config-values.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: cli/config
+  description: Don't crash on empty config values

--- a/sdk/go/common/resource/config/object.go
+++ b/sdk/go/common/resource/config/object.go
@@ -547,6 +547,8 @@ func (c object) toDecryptedPropertyValue(decrypter Decrypter) resource.PropertyV
 			values[resource.PropertyKey(k)] = v.toDecryptedPropertyValue(decrypter)
 		}
 		prop = resource.NewObjectProperty(values)
+	case nil:
+		prop = resource.NewNullProperty()
 	default:
 		contract.Failf("unexpected value type %T", v)
 	}

--- a/sdk/go/common/resource/config/object_test.go
+++ b/sdk/go/common/resource/config/object_test.go
@@ -1,3 +1,17 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package config
 
 import (

--- a/sdk/go/common/resource/config/object_test.go
+++ b/sdk/go/common/resource/config/object_test.go
@@ -1,0 +1,19 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEmptyObject(t *testing.T) {
+	t.Parallel()
+
+	// Test that an empty object can be converted to a property value
+	// without error.
+	o := object{}
+	crypter := nopCrypter{}
+	v := o.toDecryptedPropertyValue(crypter)
+	assert.Equal(t, resource.NewNullProperty(), v)
+}

